packages/odyssey-react-mui/src/labs/Switch.tsx

```diff
@@ -84,89 +87,120 @@ const StyledSwitchLabel = styled(FormLabel, {
+  };
+});
 
 const SwitchThumb = styled("span", {
   shouldForwardProp: (prop) => !nonForwardedProps.includes(prop),
 })<
-  Pick<SwitchProps, "isChecked" | "isDisabled"> & {
+  Pick<SwitchProps, "isChecked" | "isDisabled" | "isReadOnly"> & {
     odysseyDesignTokens: DesignTokens;
   }
->(({ isChecked, isDisabled, odysseyDesignTokens }) => {
+>(({ isChecked, isDisabled, isReadOnly, odysseyDesignTokens }) => {
   const thumbOffset = toRem(3);
   const trackWidth = stripRem(odysseyDesignTokens.Spacing7);
   const thumbWidth = stripRem(odysseyDesignTokens.Spacing4) - toRem(2);
-
   const transformDistance = trackWidth - thumbWidth - thumbOffset * 2;
 
+  const getBackgroundColor = () => {
+    if (isDisabled) return odysseyDesignTokens.HueNeutral50;
+    if (isReadOnly) return odysseyDesignTokens.HueNeutral400;
+    return odysseyDesignTokens.HueNeutralWhite;
+  };
+
+  const getTransform = () => {
+    if (isChecked) return `translate3d(${transformDistance}rem, -50%, 0)`;
+    return "translate3d(0, -50%, 0)";
+  };
+
   return {
     position: "absolute",
     top: "50%",
     left: `${thumbOffset}rem`,
     width: `calc(${odysseyDesignTokens.Spacing4} - ${toRem(2)}rem)`,
     height: `calc(${odysseyDesignTokens.Spacing4} - ${toRem(2)}rem)`,
     borderRadius: odysseyDesignTokens.BorderRadiusRound,
-    backgroundColor: isDisabled
-      ? odysseyDesignTokens.HueNeutral50
-      : odysseyDesignTokens.HueNeutralWhite,
-    transform: isChecked
-      ? `translate3d(${transformDistance}rem, -50%, 0)`
-      : "translate3d(0, -50%, 0)",
+    backgroundColor: getBackgroundColor(),
+    transform: getTransform(),
     transition: `transform ${odysseyDesignTokens.TransitionDurationMain}`,
   };
 });
 
 const SwitchCheckMark = styled(CheckIcon, {
   shouldForwardProp: (prop) => !nonForwardedProps.includes(prop),
 })<
-  Pick<SwitchProps, "isChecked" | "isDisabled"> & {
+  Pick<SwitchProps, "isChecked" | "isDisabled" | "isReadOnly"> & {
     odysseyDesignTokens: DesignTokens;
   }
->(({ isChecked, isDisabled, odysseyDesignTokens }) => ({
-  position: "absolute",
-  top: "50%",
-  left: 3,
-  width: odysseyDesignTokens.Spacing4,
-  transform: "translateY(-50%)",
-  transition: `opacity ${odysseyDesignTokens.TransitionDurationMain}`,
-  opacity: isChecked ? 1 : 0,
-  path: {
-    fill: isDisabled
-      ? odysseyDesignTokens.HueNeutral50
-      : odysseyDesignTokens.HueNeutralWhite,
-  },
-}));
+>(({ isChecked, isDisabled, isReadOnly, odysseyDesignTokens }) => {
+  const getOpacity = () => {
+    return isChecked ? 1 : 0;
+  };
+
+  const getPathFill = () => {
+    if (isDisabled || isReadOnly) {
+      return odysseyDesignTokens.HueNeutral50;
+    }
+    return odysseyDesignTokens.HueNeutralWhite;
+  };
 
+  return {
+    position: "absolute",
+    top: "50%",
+    left: 3,
+    width: odysseyDesignTokens.Spacing4,
+    transform: "translateY(-50%)",
+    transition: `opacity ${odysseyDesignTokens.TransitionDurationMain}`,
+    opacity: getOpacity(),
```

Comment:
If we are going to make these functions, we should let them take the args they need and move them our of the styled component declaration probably. 
```js
const getOpacity = (isChecked) => isChecked ? 1 : 0;
```
Or, we conditionally add the styles like 
```css
opacity: 0,

...(isChecked && {
  opacity: 1,
})
```

Reply:
good call, it might be an over optimization to use the functions, I'll switch it to conditionals